middleman-core/features/cli/preview_server.feature

```diff
@@ -0,0 +1,376 @@
+Feature: Run the preview server
+
+  As a software developer
+  I want to start the preview server
+  In order to view my changes immediately in the browser
+
+  Background:
+    Given a fixture app "clean-app"
+    And the default aruba timeout is 15 seconds
+    And I set the environment variables to:
+      | variable | value  |
+      | HOSTSRC  | .hosts |
+    And a file named ".hosts" with:
+    """
+    # <ip-address> <hostname.domain.org> <hostname>
+    127.0.0.1 localhost.localdomain localhost
+    """
+
+  Scenario: Start the server with defaults
+    When I run `middleman server` interactively
+    And I stop middleman if the output contains:
+    """
+    Inspect your site configuration
+    """
+    Then the output should contain:
+    """
+    The Middleman is listening on "http://:::4567", "http://0.0.0.0:4567"
+    """
+    And the output should contain:
+    """
+    View your site at "http://
+    """
+    And the output should contain:
+    """
+    Inspect your site configuration at "http://
+    """
+
+  Scenario: Start the server with bind address 127.0.0.1
+    When I run `middleman server --bind-address 127.0.0.1` interactively
+    And I stop middleman if the output contains:
+    """
+    Inspect your site configuration
+    """
+    Then the output should contain:
+    """
+    The Middleman is listening on "http://127.0.0.1:4567"
+    """
+    And the output should contain:
+    """
+    View your site at "http://127.0.0.1:4567"
+    """
+    And the output should contain:
+    """
+    Inspect your site configuration at "http://127.0.0.1:4567/__middleman"
+    """
+
+  Scenario: Start the server with bind address 127.0.0.1 configured via config.rb
+    Given a file named "config.rb" with:
+    """
+    set :bind_address, '127.0.0.1'
+    """
+    When I run `middleman server` interactively
+    And I stop middleman if the output contains:
+    """
+    Inspect your site configuration
+    """
+    Then the output should contain:
+    """
+    The Middleman is listening on "http://127.0.0.1:4567"
+    """
+    And the output should contain:
+    """
+    View your site at "http://127.0.0.1:4567"
+    """
+    And the output should contain:
+    """
+    Inspect your site configuration at "http://127.0.0.1:4567/__middleman"
```

Comment:
Mmmh... But then the user cannot Copy&Paste the address to his/her browser. Would like to leave it like that.